relay_test.go

```diff
@@ -384,3 +384,55 @@ func TestRelayOutgoingConnectionsEphemeral(t *testing.T) {
 		require.NoError(t, testutils.CallEcho(ts.Server(), ts.HostPort(), "s2", nil), "CallEcho failed")
 	})
 }
+
+func TestRelayHandleLocalCall(t *testing.T) {
+	opts := testutils.NewOpts().SetRelayOnly().
+		SetRelayLocal("relay", "tchannel", "test").
+		// We make a call to "test" for an unknown method.
+		AddLogFilter("Couldn't find handler.", 1)
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		s2 := ts.NewServer(serviceNameOpts("s2"))
+		testutils.RegisterEcho(s2, nil)
+
+		client := ts.NewClient(nil)
+		testutils.AssertEcho(t, client, ts.HostPort(), "s2")
+
+		testutils.RegisterEcho(ts.Relay(), nil)
+		testutils.AssertEcho(t, client, ts.HostPort(), "relay")
+
+		// Sould get a bad request for "test" since the channel does not handle it.
+		err := testutils.CallEcho(client, ts.HostPort(), "test", nil)
+		assert.Equal(t, ErrCodeBadRequest, GetSystemErrorCode(err), "Expected BadRequest for test")
+
+		// But an unknown service causes declined
+		err = testutils.CallEcho(client, ts.HostPort(), "unknown", nil)
+		assert.Equal(t, ErrCodeDeclined, GetSystemErrorCode(err), "Expected Declined for unknown")
+	})
+}
+
+func TestRelayHandleLargeLocalCall(t *testing.T) {
+	opts := testutils.NewOpts().SetRelayOnly().
+		SetRelayLocal("relay").
+		AddLogFilter("Received fragmented callReq", 1).
+		// Expect 4 callReqContinues for 256 kb payload that we cannot relay.
+		AddLogFilter("Failed to relay frame.", 4)
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		client := ts.NewClient(nil)
+		testutils.RegisterEcho(ts.Relay(), nil)
+
+		// This large call should fail with a bad request.
+		err := testutils.CallEcho(client, ts.HostPort(), "relay", &raw.Args{
+			Arg2: testutils.RandBytes(128 * 1024),
+			Arg3: testutils.RandBytes(128 * 1024),
+		})
+		if assert.Equal(t, ErrCodeBadRequest, GetSystemErrorCode(err), "Expected BadRequest for large call to relay") {
+			assert.Contains(t, err.Error(), "cannot receive fragmented calls")
+		}
+
+		// We may get an error before the call is finished flushing.
+		// Do a ping to ensure everything has been flushed.
+		ctx, cancel := NewContext(time.Second)
```

Comment:
Any test that uses low timeouts (like under 100 milliseconds) uses the timeout function, but anything with a large timeout (like `time.Second`) hasn't been using it in tests (we have a whole bunch of these littered throughout the code). I'd rather keep this consistent with what the other tests do.

In future, I'd like to add a `TestServer.NewContext` which will automatically `defer` at the end of the test so that callers like this could inline if they wanted. Maybe a `NewDefaultContext` which will make a context with some timeout that is scaled automatically.